src/apps/engine/src/compiler.cpp

```diff
@@ -1197,11 +1213,17 @@ bool COMPILER::Compile(SEGMENT_DESC &Segment, char *pInternalCode, uint32_t pInt
 
     if (pInternalCode == nullptr)
     {
-        Segment.Files_list->AddUnicalString(file_name);
+        auto is_new = Segment.Files_list->AddUnicalString(file_name);
         file_code = Segment.Files_list->GetStringCode(file_name);
         pProgram = nullptr;
         Program_size = 0;
         pSegmentSource = LoadFile(file_name, SegmentSize);
+        if (is_new && use_script_cache_)
+        {
+            script_cache_.files.emplace_back(file_name);
+            script_cache_.crc = storm::script_cache::ComputeCRC(0, {});
```

Comment:
why this is needed? it seems to make no difference

Reply:
fixed